File: source/clibCore/CPUserClass.vb

Imports Contensive.BaseClasses
Imports System.Runtime.InteropServices

Namespace Contensive.Core
    '
    ' comVisible to be activeScript compatible
    '
    <ComVisible(True)> _
    <ComClass(CPUserClass.ClassId, CPUserClass.InterfaceId, CPUserClass.EventsId)> _
    Public Class CPUserClass
        Inherits BaseClasses.CPUserBaseClass
        Implements IDisposable
        '
#Region "COM GUIDs"
        Public Const ClassId As String = "08EF64C6-9C51-4B32-84D9-0D3BDAF42A28"
        Public Const InterfaceId As String = "B1A95B1F-A00D-4AC6-B3A0-B1619568C2EA"
        Public Const EventsId As String = "DBE2B6CB-6339-4FFB-92D7-BE37AEA841CC"
#End Region
        '
        Private cpCore As Contensive.Core.coreClass
        Private CP As CPClass
        Protected disposed As Boolean = False
        '
        '====================================================================================================
        '
        Public Sub New(ByVal cpCoreObj As Contensive.Core.coreClass, ByVal CPParent As CPClass)
            MyBase.New()
            cpCore = cpCoreObj
            CP = CPParent
        End Sub
        '
        '====================================================================================================
        '
        Protected Overridable Overloads Sub Dispose(ByVal disposing As Boolean)
            If Not Me.disposed Then
                Call appendDebugLog(".dispose, dereference cp, main, csv")
                If disposing Then
                    '
                    ' call .dispose for managed objects
                    '
                    CP = Nothing
                    cpCore = Nothing
                End If
                '
                ' Add code here to release the unmanaged resource.
                '
            End If
            Me.disposed = True
        End Sub
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property Email() As String 'Inherits BaseClasses.CPUserBaseClass.Email
            Get
                If True Then
                    Return cpCore.user.email
                Else
                    Return ""
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides Function GetIdByLogin(ByVal Username As String, ByVal Password As String) As Integer 'Inherits BaseClasses.CPUserBaseClass.GetIdByLogin
            If True Then
                Return cpCore.user.authenticateGetId(Username, Password)
            Else
                Return 0
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property Id() As Integer 'Inherits BaseClasses.CPUserBaseClass.Id
            Get
                Dim localId As Integer = 0
                '
                If True Then
                    localId = cpCore.user.id
                    If (localId = 0) Then
                        localId = cpCore.db.metaData_InsertContentRecordGetID("people", 0)
                        Call cpCore.user.recognizeById(localId)
                    End If
                End If
                Return localId
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property IsAdmin() As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsAdmin
            Get
                If True Then
                    Return cpCore.user.isAuthenticatedAdmin
                Else
                    Return False
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides Function IsAdvancedEditing(ByVal ContentName As String) As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsAdvancedEditing
            If True Then
                Return cpCore.user.isAdvancedEditing(ContentName)
            Else
                Return False
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property IsAuthenticated() As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsAuthenticated
            Get
                If True Then
                    Return (cpCore.user.isAuthenticated())
                Else
                    Return False
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides Function IsAuthoring(ByVal ContentName As String) As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsAuthoring
            If True Then
                Return cpCore.user.isEditing(ContentName)
            Else
                Return False
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Overrides Function IsContentManager(Optional ByVal ContentName As String = "Page Content") As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsContentManager
            If True Then
                Return cpCore.user.isAuthenticatedContentManager(ContentName)
            Else
                Return False
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property IsDeveloper() As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsDeveloper
            Get
                If True Then
                    Return cpCore.user.isAuthenticatedDeveloper()
                Else
                    Return False
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides Function IsEditing(ByVal ContentName As String) As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsEditing
            If True Then
                Return cpCore.user.isEditing(ContentName)
            Else
                Return False
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property IsEditingAnything() As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsEditingAnything
            Get
                If True Then
                    Return cpCore.user.isEditingAnything()
                Else
                    Return False
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property IsGuest() As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsGuest
            Get
                If True Then
                    Return cpCore.user.isGuest
                Else
                    Return False
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides Function IsInGroup(ByVal groupName As String, Optional ByVal userId As Integer = 0) As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsInGroup
            Dim groupId As Integer
            Dim result As Boolean
            '
            Try
                groupId = CP.Group.GetId(groupName)
                If userId = 0 Then
                    userId = Id
                End If
                If groupId = 0 Then
                    result = False
                Else
                    result = IsInGroupList(groupId.ToString, userId)
                End If
            Catch ex As Exception
                Call CP.core.handleExceptionAndRethrow(ex, "Unexpected error in cs.user.IsInGroup")
                result = False
            End Try
            Return result
        End Function
        '
        '====================================================================================================
        '
        Public Overrides Function IsInGroupList(ByVal GroupIDList As String, Optional ByVal userId As Integer = 0) As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsInGroup
            Dim result As Boolean
            '
            Try
                If userId = 0 Then
                    userId = Id
                End If
                result = cpCore.user.isMemberOfGroupIdList(userId, IsAuthenticated(), GroupIDList, False)
            Catch ex As Exception
                Call CP.core.handleExceptionAndRethrow(ex, "Unexpected error in cs.user.IsInGroupList")
                result = False
            End Try
            Return result
            '
        End Function
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property IsMember() As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsMember
            Get
                If True Then
                    Return cpCore.user.isAuthenticatedMember
                Else
                    Return False
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides Function IsQuickEditing(ByVal ContentName As String) As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsQuickEditing
            If True Then
                Return cpCore.user.isQuickEditing(ContentName)
            Else
                Return False
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property IsRecognized() As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsRecognized
            Get
                If True Then
                    Return cpCore.user.isRecognized
                Else
                    Return False
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property IsWorkflowRendering() As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsWorkflowRendering
            Get
                If True Then
                    Return cpCore.pagemanager_IsWorkflowRendering
                Else
                    Return False
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property Language() As String 'Inherits BaseClasses.CPUserBaseClass.Language
            Get
                If True Then
                    Return cpCore.user.language
                Else
                    Return ""
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property LanguageID() As Integer 'Inherits BaseClasses.CPUserBaseClass.LanguageId
            Get
                If True Then
                    Return cpCore.user.languageId
                Else
                    Return 0
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides Function Login(ByVal UsernameOrEmail As String, ByVal Password As String, Optional ByVal SetAutoLogin As Boolean = False) As Boolean 'Inherits BaseClasses.CPUserBaseClass.Login
            If True Then
                Return cpCore.user.authenticate(UsernameOrEmail, Password, SetAutoLogin)
            Else
                Return False
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Overrides Function LoginByID(ByVal RecordID As String, Optional ByVal SetAutoLogin As Boolean = False) As Boolean 'Inherits BaseClasses.CPUserBaseClass.LoginById
            If True Then
                Return cpCore.user.authenticateById(RecordID, SetAutoLogin)
            Else
                Return False
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Overrides Function LoginIsOK(ByVal UsernameOrEmail As String, ByVal Password As String) As Boolean 'Inherits BaseClasses.CPUserBaseClass.LoginIsOK
            If True Then
                Return cpCore.main_IsLoginOK(UsernameOrEmail, Password)
            Else
                Return False
            End If
        End Function
        '
        '====================================================================================================
        '
        Public Overrides Sub Logout() 'Inherits BaseClasses.CPUserBaseClass.Logout
            Call cpCore.user.logout()
        End Sub
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property Name() As String 'Inherits BaseClasses.CPUserBaseClass.Name
            Get
                If True Then
                    Return cpCore.user.name
                Else
                    Return ""
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property IsNew() As Boolean 'Inherits BaseClasses.CPUserBaseClass.IsNew
            Get
                Return cpCore.user.isNew
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides Function IsNewLoginOK(ByVal Username As String, ByVal Password As String) As Boolean 'Inherits BaseClasses.CPUserBaseClass.NewLoginIsOK
            Dim errorMessage As String = ""
            Dim errorCode As Integer = 0
            Return cpCore.user.isNewLoginOK(Username, Password, errorMessage, errorCode)
        End Function
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property OrganizationID() As Integer 'Inherits BaseClasses.CPUserBaseClass.OrganizationId
            Get
                If True Then
                    Return cpCore.user.organizationId
                Else
                    Return 0
                End If
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property Password() As String 'Inherits BaseClasses.CPUserBaseClass.Password
            Get
                Return cpCore.user.password
            End Get
        End Property
        '
        '====================================================================================================
        '
        Public Overrides Function Recognize(ByVal UserID As Integer) As Boolean 'Inherits BaseClasses.CPUserBaseClass.Recognize
            Return cpCore.user.recognizeById(UserID)
        End Function
        '
        '====================================================================================================
        '
        Public Overrides ReadOnly Property Username() As String 'Inherits BaseClasses.CPUserBaseClass.Username
            Get
                Return cpCore.user.username
            End Get
        End Property
        '
        '=======================================================================================================
        '
        Public Overrides Function GetProperty(ByVal PropertyName As String, Optional ByVal DefaultValue As String = "", Optional ByVal TargetMemberId As Integer = 0) As String
            If (TargetMemberId = 0) Then
                Return cpCore.userProperty.getText(PropertyName, DefaultValue)
            Else
                Return cpCore.userProperty.getText(PropertyName, DefaultValue, TargetMemberId)
            End If
        End Function
        '
        '=======================================================================================================
        '
        Public Overrides Sub SetProperty(ByVal PropertyName As String, ByVal Value As String, Optional ByVal TargetMemberId As Integer = 0)
            If (TargetMemberId = 0) Then
                Call cpCore.userProperty.setProperty(PropertyName, Value)
            Else
                Call cpCore.userProperty.setProperty(PropertyName, Value, TargetMemberId)
            End If
        End Sub
        '
        '=======================================================================================================
        ' REFACTOR -- obsolete this and setup the defaultValue type correctly, and add the targetUserId
        '=======================================================================================================
        '
        Public Overrides Function GetBoolean(PropertyName As String, Optional DefaultValue As String = "") As Boolean
            Return cpCore.userProperty.getBoolean(PropertyName, DefaultValue)
        End Function
        '
        '=======================================================================================================
        ' REFACTOR -- obsolete this and setup the defaultValue type correctly, and add the targetUserId
        '=======================================================================================================
        '
        Public Overrides Function GetDate(PropertyName As String, Optional DefaultValue As String = "") As Date
            Return cpCore.userProperty.getDate(PropertyName, DefaultValue)
        End Function
        '
        '=======================================================================================================
        ' REFACTOR -- obsolete this and setup the defaultValue type correctly, and add the targetUserId
        '=======================================================================================================
        '
        Public Overrides Function GetInteger(PropertyName As String, Optional DefaultValue As String = "") As Integer
            Return cpCore.userProperty.getInteger(PropertyName, DefaultValue)
        End Function
        '
        '=======================================================================================================
        ' REFACTOR -- obsolete this and setup the defaultValue type correctly, and add the targetUserId
        '=======================================================================================================
        '
        Public Overrides Function GetNumber(PropertyName As String, Optional DefaultValue As String = "") As Double
            Return cpCore.userProperty.getNumber(PropertyName, DefaultValue)
        End Function
        '
        '=======================================================================================================
        '
        Public Overrides Function GetText(PropertyName As String, Optional DefaultValue As String = "") As String
            Return cpCore.userProperty.getText(PropertyName, DefaultValue)
        End Function
        '
        '====================================================================================================
        ' REFACTOR -- figure out what this does and rewrite
        '
        Public Overrides Sub track()
            Dim localId As Integer
            If True Then
                '
                ' get the id property, which triggers a track() if it returns o
                '
                localId = Id
                'If Id = 0 Then
                '    localId = cmc.csv_InsertContentRecordGetID("people", 0)
                '    Call cmc.main_RecognizeMemberByID(localId)
                'End If
            End If
        End Sub
        '
        '====================================================================================================
        '
        Private Sub appendDebugLog(ByVal copy As String)
            '
        End Sub
        '
        '====================================================================================================
        '
        Private Sub tp(ByVal msg As String)
            'Call appendDebugLog(msg)
        End Sub
#Region " IDisposable Support "
        ' Do not change or add Overridable to these methods.
        ' Put cleanup code in Dispose(ByVal disposing As Boolean).
        Public Overloads Sub Dispose() Implements IDisposable.Dispose
            Dispose(True)
            GC.SuppressFinalize(Me)
        End Sub
        Protected Overrides Sub Finalize()
            Dispose(False)
            MyBase.Finalize()
        End Sub
#End Region
    End Class

End Namespace

File: source/clibCoreCs/Interfaces/cpApi/CPUserClass.cs
using System;

//========================================================================
// This conversion was produced by the Free Edition of
// Instant C# courtesy of Tangible Software Solutions.
// Order the Premium Edition at https://www.tangiblesoftwaresolutions.com
//========================================================================

using Contensive.Core.Controllers;

using Contensive.BaseClasses;
using System.Runtime.InteropServices;

namespace Contensive.Core
{
	//
	// comVisible to be activeScript compatible
	//
	[ComVisible(true), Microsoft.VisualBasic.ComClass(CPUserClass.ClassId, CPUserClass.InterfaceId, CPUserClass.EventsId)]
	public class CPUserClass : BaseClasses.CPUserBaseClass, IDisposable
	{
		//
#region COM GUIDs
		public const string ClassId = "08EF64C6-9C51-4B32-84D9-0D3BDAF42A28";
		public const string InterfaceId = "B1A95B1F-A00D-4AC6-B3A0-B1619568C2EA";
		public const string EventsId = "DBE2B6CB-6339-4FFB-92D7-BE37AEA841CC";
#endregion
		//
		private Contensive.Core.coreClass cpCore;
		private CPClass CP;
		protected bool disposed = false;
		//
		//====================================================================================================
		//
		public CPUserClass(Contensive.Core.coreClass cpCoreObj, CPClass CPParent) : base()
		{
			cpCore = cpCoreObj;
			CP = CPParent;
		}
		//
		//====================================================================================================
		//
		protected virtual void Dispose(bool disposing)
		{
			if (!this.disposed)
			{
				appendDebugLog(".dispose, dereference cp, main, csv");
				if (disposing)
				{
					//
					// call .dispose for managed objects
					//
					CP = null;
					cpCore = null;
				}
				//
				// Add code here to release the unmanaged resource.
				//
			}
			this.disposed = true;
		}
		//
		//====================================================================================================
		//
		public override string Email //Inherits BaseClasses.CPUserBaseClass.Email
		{
			get
			{
				if (true)
				{
					return CP.core.doc.authContext.user.Email;
				}
				else
				{
					return "";
				}
			}
		}
		//
		//====================================================================================================
		//
		public override int GetIdByLogin(string Username, string Password) //Inherits BaseClasses.CPUserBaseClass.GetIdByLogin
		{
			if (true)
			{
				return CP.core.doc.authContext.authenticateGetId(cpCore, Username, Password);
			}
			else
			{
				return 0;
			}
		}
		//
		//====================================================================================================
		//
		public override int Id //Inherits BaseClasses.CPUserBaseClass.Id
		{
			get
			{
				int localId = 0;
				//
				if (true)
				{
					localId = CP.core.doc.authContext.user.id;
					if (localId == 0)
					{
						localId = CP.core.db.insertContentRecordGetID("people", 0);
						CP.core.doc.authContext.recognizeById(cpCore, localId, CP.core.doc.authContext);
					}
				}
				return localId;
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsAdmin //Inherits BaseClasses.CPUserBaseClass.admin
		{
			get
			{
				if (true)
				{
					return CP.core.doc.authContext.isAuthenticatedAdmin(cpCore);
				}
				else
				{
					return false;
				}
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsAdvancedEditing(string ContentName) //Inherits BaseClasses.CPUserBaseClass.IsAdvancedEditing
		{
			if (true)
			{
				return CP.core.doc.authContext.isAdvancedEditing(cpCore, ContentName);
			}
			else
			{
				return false;
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsAuthenticated //Inherits BaseClasses.CPUserBaseClass.IsAuthenticated
		{
			get
			{
				if (true)
				{
					return (CP.core.doc.authContext.isAuthenticated);
				}
				else
				{
					return false;
				}
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsAuthoring(string ContentName) //Inherits BaseClasses.CPUserBaseClass.IsAuthoring
		{
			if (true)
			{
				return CP.core.doc.authContext.isEditing(ContentName);
			}
			else
			{
				return false;
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsContentManager(string ContentName = "Page Content") //Inherits BaseClasses.CPUserBaseClass.IsContentManager
		{
			if (true)
			{
				return CP.core.doc.authContext.isAuthenticatedContentManager(cpCore, ContentName);
			}
			else
			{
				return false;
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsDeveloper //Inherits BaseClasses.CPUserBaseClass.IsDeveloper
		{
			get
			{
				if (true)
				{
					return CP.core.doc.authContext.isAuthenticatedDeveloper(cpCore);
				}
				else
				{
					return false;
				}
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsEditing(string ContentName) //Inherits BaseClasses.CPUserBaseClass.IsEditing
		{
			if (true)
			{
				return CP.core.doc.authContext.isEditing(ContentName);
			}
			else
			{
				return false;
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsEditingAnything //Inherits BaseClasses.CPUserBaseClass.IsEditingAnything
		{
			get
			{
				if (true)
				{
					return CP.core.doc.authContext.isEditingAnything();
				}
				else
				{
					return false;
				}
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsGuest //Inherits BaseClasses.CPUserBaseClass.IsGuest
		{
			get
			{
				if (true)
				{
					return CP.core.doc.authContext.isGuest(cpCore);
				}
				else
				{
					return false;
				}
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsInGroup(string groupName, int userId = 0) //Inherits BaseClasses.CPUserBaseClass.IsInGroup
		{
			int groupId = 0;
			bool result = false;
			//
			try
			{
				groupId = CP.Group.GetId(groupName);
				if (userId == 0)
				{
					userId = Id;
				}
				if (groupId == 0)
				{
					result = false;
				}
				else
				{
					result = IsInGroupList(groupId.ToString(), userId);
				}
			}
			catch (Exception ex)
			{
				CP.core.handleException(ex, "Unexpected error in cs.user.IsInGroup");
				result = false;
			}
			return result;
		}
		//
		//====================================================================================================
		//
		public override bool IsInGroupList(string GroupIDList, int userId = 0) //Inherits BaseClasses.CPUserBaseClass.IsInGroup
		{
			bool result = false;
			//
			try
			{
				if (userId == 0)
				{
					userId = Id;
				}
				result = CP.core.doc.authContext.isMemberOfGroupIdList(cpCore, userId, IsAuthenticated, GroupIDList, false);
			}
			catch (Exception ex)
			{
				CP.core.handleException(ex, "Unexpected error in cs.user.IsInGroupList");
				result = false;
			}
			return result;
			//
		}
		//
		//====================================================================================================
		//
		public override bool IsMember //Inherits BaseClasses.CPUserBaseClass.IsMember
		{
			get
			{
				if (true)
				{
					return CP.core.doc.authContext.isAuthenticatedMember(cpCore);
				}
				else
				{
					return false;
				}
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsQuickEditing(string ContentName) //Inherits BaseClasses.CPUserBaseClass.IsQuickEditing
		{
			return CP.core.doc.authContext.isQuickEditing(cpCore, ContentName);
		}
		//
		//====================================================================================================
		//
		public override bool IsRecognized //Inherits BaseClasses.CPUserBaseClass.IsRecognized
		{
			get
			{
				return CP.core.doc.authContext.isRecognized(cpCore);
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsWorkflowRendering //Inherits BaseClasses.CPUserBaseClass.IsWorkflowRendering
		{
			get
			{
				return CP.core.doc.authContext.isWorkflowRendering;
			}
		}
		//
		//====================================================================================================
		//
		public override string Language //Inherits BaseClasses.CPUserBaseClass.Language
		{
			get
			{
				if (CP.core.doc.authContext.userLanguage != null)
				{
					return CP.core.doc.authContext.userLanguage.name;
				}
				return string.Empty;
			}
		}
		//
		//====================================================================================================
		//
		public override int LanguageID //Inherits BaseClasses.CPUserBaseClass.LanguageId
		{
			get
			{
				return CP.core.doc.authContext.user.LanguageID;
			}
		}
		//
		//====================================================================================================
		//
		public override bool Login(string UsernameOrEmail, string Password, bool SetAutoLogin = false) //Inherits BaseClasses.CPUserBaseClass.Login
		{
			return CP.core.doc.authContext.authenticate(cpCore, UsernameOrEmail, Password, SetAutoLogin);
		}
		//
		//====================================================================================================
		//
		[Obsolete("Use LoginById(integer) instead", false)]
		public override bool LoginByID(string RecordID, bool SetAutoLogin = false)
		{
			return CP.core.doc.authContext.authenticateById(cpCore, EncodeInteger(RecordID), CP.core.doc.authContext);
		}
		//
		//====================================================================================================
		//
		public override bool LoginByID(int RecordID)
		{
			return CP.core.doc.authContext.authenticateById(cpCore, RecordID, CP.core.doc.authContext);
		}
		//
		//====================================================================================================
		//
		public override bool LoginByID(int RecordID, bool SetAutoLogin)
		{
			return CP.core.doc.authContext.authenticateById(cpCore, RecordID, CP.core.doc.authContext);
			if (!CP.core.doc.authContext.user.AutoLogin)
			{
				CP.core.doc.authContext.user.AutoLogin = true;
				CP.core.doc.authContext.user.save(cpCore);
			}
//INSTANT C# NOTE: Inserted the following 'return' since all code paths must return a value in C#:
			return false;
		}
		//
		//====================================================================================================
		//
		public override bool LoginIsOK(string UsernameOrEmail, string Password)
		{
			return CP.core.doc.authContext.isLoginOK(cpCore, UsernameOrEmail, Password);
		}
		//
		//====================================================================================================
		//
		public override void Logout()
		{
			CP.core.doc.authContext.logout(cpCore);
		}
		//
		//====================================================================================================
		//
		public override string Name
		{
			get
			{
				return CP.core.doc.authContext.user.name;
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsNew
		{
			get
			{
				return CP.core.doc.authContext.visit.MemberNew;
			}
		}
		//
		//====================================================================================================
		//
		public override bool IsNewLoginOK(string Username, string Password)
		{
			string errorMessage = "";
			int errorCode = 0;
			return CP.core.doc.authContext.isNewLoginOK(cpCore, Username, Password, errorMessage, errorCode);
		}
		//
		//====================================================================================================
		//
		public override int OrganizationID
		{
			get
			{
				return CP.core.doc.authContext.user.OrganizationID;
			}
		}
		//
		//====================================================================================================
		//
		public override string Password //Inherits BaseClasses.CPUserBaseClass.Password
		{
			get
			{
				return CP.core.doc.authContext.user.Password;
			}
		}
		//
		//====================================================================================================
		//
		public override bool Recognize(int UserID) //Inherits BaseClasses.CPUserBaseClass.Recognize
		{
			return CP.core.doc.authContext.recognizeById(cpCore, UserID, CP.core.doc.authContext);
		}
		//
		//====================================================================================================
		//
		public override string Username //Inherits BaseClasses.CPUserBaseClass.Username
		{
			get
			{
				return CP.core.doc.authContext.user.Username;
			}
		}
		//
		//=======================================================================================================
		//
		public override string GetProperty(string PropertyName, string DefaultValue = "", int TargetMemberId = 0)
		{
			if (TargetMemberId == 0)
			{
				return CP.core.userProperty.getText(PropertyName, DefaultValue);
			}
			else
			{
				return CP.core.userProperty.getText(PropertyName, DefaultValue, TargetMemberId);
			}
		}
		//
		//=======================================================================================================
		//
		public override void SetProperty(string PropertyName, string Value, int TargetMemberId = 0)
		{
			if (TargetMemberId == 0)
			{
				CP.core.userProperty.setProperty(PropertyName, Value);
			}
			else
			{
				CP.core.userProperty.setProperty(PropertyName, Value, TargetMemberId);
			}
		}
		//
		//=======================================================================================================
		// REFACTOR -- obsolete this and setup the defaultValue type correctly, and add the targetUserId
		//=======================================================================================================
		//
		public override bool GetBoolean(string PropertyName, string DefaultValue = "")
		{
			return CP.core.userProperty.getBoolean(PropertyName, genericController.EncodeBoolean(DefaultValue));
		}
		//
		//=======================================================================================================
		// REFACTOR -- obsolete this and setup the defaultValue type correctly, and add the targetUserId
		//=======================================================================================================
		//
		public override DateTime GetDate(string PropertyName, string DefaultValue = "")
		{
			return CP.core.userProperty.getDate(PropertyName, genericController.EncodeDate(DefaultValue));
		}
		//
		//=======================================================================================================
		// REFACTOR -- obsolete this and setup the defaultValue type correctly, and add the targetUserId
		//=======================================================================================================
		//
		public override int GetInteger(string PropertyName, string DefaultValue = "")
		{
			return CP.core.userProperty.getInteger(PropertyName, genericController.EncodeInteger(DefaultValue));
		}
		//
		//=======================================================================================================
		// REFACTOR -- obsolete this and setup the defaultValue type correctly, and add the targetUserId
		//=======================================================================================================
		//
		public override double GetNumber(string PropertyName, string DefaultValue = "")
		{
			return CP.core.userProperty.getNumber(PropertyName, EncodeNumber(DefaultValue));
		}
		//
		//=======================================================================================================
		//
		public override string GetText(string PropertyName, string DefaultValue = "")
		{
			return CP.core.userProperty.getText(PropertyName, DefaultValue);
		}
		//
		//====================================================================================================
		// REFACTOR -- figure out what this does and rewrite
		//
		public override void track()
		{
			int localId = 0;
			if (true)
			{
				//
				// get the id property, which triggers a track() if it returns o
				//
				localId = Id;
				//If Id = 0 Then
				//    localId = cmc.csv_InsertContentRecordGetID("people", 0)
				//    Call cmc.main_RecognizeMemberByID(localId)
				//End If
			}
		}
		//
		//====================================================================================================
		//
		private void appendDebugLog(string copy)
		{
			//
		}
		//
		//====================================================================================================
		//
		private void tp(string msg)
		{
			//Call appendDebugLog(msg)
		}
#region  IDisposable Support 
		// Do not change or add Overridable to these methods.
		// Put cleanup code in Dispose(ByVal disposing As Boolean).
		public void Dispose()
		{
			Dispose(true);
			GC.SuppressFinalize(this);
		}
		~CPUserClass()
		{
			Dispose(false);
//INSTANT C# NOTE: The base class Finalize method is automatically called from the destructor:
			//base.Finalize();
		}
#endregion
	}

}